File: problems/438/solution.go
package main
import "fmt"

func findAnagrams(s string, p string) []int {
    chCount, ans := make([]int, 26), []int{}
    
    for _, ch := range p {
        chCount[ch - 'a']++
    }
    
    left, right, count := 0, 0, len(p)
    
    for right < len(s) {
        if chCount[s[right] - 'a'] >= 1 {
            count--
        }
        
        chCount[s[right] - 'a']--
        right++
        
        if count == 0 {
            ans = append(ans, left)
        }
        
        if right - left == len(p) {
            if chCount[s[left] - 'a'] >= 0 {
                count++
            }
            
            chCount[s[left]  - 'a']++
            left++
        }
    }
    
    return ans
}

File: problems/438/solution.py
class Solution:
    def findAnagrams(self, s: str, p: str) -> List[int]:
        if not s:
            return []
        
        chCounter, ans = [0] * 26, []
        
        for ch in p:
            chCounter[ord(ch) - ord('a')] += 1
        
        left, right, count = 0, 0, len(p)
        
        while right < len(s):
            if chCounter[ord(s[right]) - ord('a')] >= 1:
                count -= 1
            
            chCounter[ord(s[right]) - ord('a')] -= 1
            right += 1
            
            if count == 0:
                ans.append(left)
            
            if right - left == len(p):
                if chCounter[ord(s[left]) - ord('a')] >= 0:
                    count += 1
                
                chCounter[ord(s[left]) - ord('a')] += 1
                left += 1
        
        return ans